packages/sysutils/busybox/scripts/fs-resize

```diff
@@ -48,8 +48,11 @@ if [ -e /storage/.please_resize_me ] ; then
       echo "Please do not reboot or turn off your @DISTRONAME@ device!"
       echo ""
 
-      # fix any minor issues, such as gpt header not at end of disk
-      StartProgress spinner "Checking layout...   " "sgdisk -e $DISK &>/dev/null"
+      # identify the partition scheme, and if gpt fix minor issues such as gpt header not at end of disk
+      SCHEME=$(parted -s -m $DISK unit b print 2>/dev/null | grep /dev | sed 's/^.*\(msdos\|gpt\).*$/\1/')
```

Comment:
`grep ^/dev/` might be safer (avoid matching on /dev anywhere else)

You could also replace the entire `sed` with `cut -d: -f6` which is much easier (tested on Generic and `-f6` works for `gpt` too).
##### RPi2:
```
rpi22:~ # parted -s -m /dev/mmcblk0 unit b print
BYT;
/dev/mmcblk0:31914983424B:sd/mmc:512:512:msdos:SD SP32G:;
1:1048576B:1074790399B:1073741824B:fat32::;
2:1074790400B:31914459135B:30839668736B:ext4::;
rpi22:~ # parted -s -m /dev/mmcblk0 unit b print | grep ^/dev/ | cut -d: -f6
msdos
```
##### Generic:
```
NUC:~ # parted -s -m /dev/sda unit b print
BYT;
/dev/sda:128035676160B:scsi:512:512:gpt:ATA SanDisk SD8SNAT1:;
1:1048576B:1075838975B:1074790400B:fat32:primary:boot, legacy_boot, esp;
2:1075838976B:128035323903B:126959484928B:ext4:primary:;
NUC:~ # parted -s -m /dev/sda unit b print | grep ^/dev/ | cut -d: -f6
gpt
```

Reply:
updated, thanks